From e1bff829a01715c7c9053b3fb8cc5ecb9d041b43 Mon Sep 17 00:00:00 2001
From: classabbyamp <5366828+classabbyamp@users.noreply.github.com>
Date: Sun, 6 Aug 2023 10:44:27 -0400
Subject: [PATCH] proto-irc.c: handle WHOX replies with realnames that aren't
 :-prefixed

the colon is not required if the last argument is a single word

this was causing issues where the user info context menu was stripping
the first character of single-word realnames when sending the WHOX
command `WHO #channel %chtsunfra,152`. Discovered the issue by seeing that
plain `WHO #channel` fixed the issue temporarily.
---
 src/common/proto-irc.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/proto-irc.c b/src/common/proto-irc.c
index 5b8e02c48..9ef25730b 100644
--- a/src/common/proto-irc.c
+++ b/src/common/proto-irc.c
@@ -762,7 +762,7 @@ process_numeric (session * sess, int n,
 
 				/* :server 354 yournick 152 #channel ~ident host servname nick H account :realname */
 				inbound_user_info (sess, word[5], word[6], word[7], word[8],
-										 word[9], word_eol[12]+1, word[11], away,
+										 word[9], word_eol[12][0] == ':' ? word_eol[12] + 1 : word_eol[12], word[11], away,
 										 tags_data);
 
 				/* try to show only user initiated whos */
